src/twisted/web/test/test_web.py

```diff
@@ -865,6 +865,19 @@ def test_noDefaultContentTypeOn204Response(self):
         self.assertNotIn(b'content-type', response.lower())
 
 
+    def test_hyperlinkRequestURL(self):
+        """
+        L{Request.prePathURL} quotes special characters in the URL segments to
```

Comment:
This test seems oddly specific -- I'd expect us to have at least one test that does an equality comparison directly from `request.url` to a manually constructed `hyperlink.URL` -- then if there's some additional quoting behavior that needs checking yeah additional test cases might be in order, but I think we need the simple one too.